source/slang/slang-lower-to-ir.cpp

```diff
@@ -6050,6 +6067,21 @@ struct StmtLoweringVisitor : StmtVisitor<StmtLoweringVisitor>
             // with no value, which can only occur in a function with
             // a `void` result type.
             //
+            // Special handling when `return` is inside a constructor.
+            if (constructorDecl)
+            {
+                if (context->returnDestination.flavor != LoweredValInfo::Flavor::None)
+                {
+                    // Simply emit `return` for NonCopyableType.
```

Comment:
When writing comments, focus on why rather than what. It is clear that the code is emitting just a return. But we should try to explain why:

A constructor for non copyable type is in the form of 
void ctor(inout this),
Where a constructor for ordinary type is in the form of
Type ctor()